lib/std/os.zig

```diff
@@ -3237,3 +3237,20 @@ pub fn sched_yield() SchedYieldError!void {
         else => return error.SystemCannotYield,
     }
 }
+
+/// Set a socket's options.
+pub fn setsockopt(fd: fd_t, level: u32, optname: u32, opt: []const u8) !void {
+    switch (errno(system.setsockopt(fd, level, optname, opt.ptr, @intCast(socklen_t, opt.len)))) {
+        0 => {},
+        EBADF => unreachable,
+        EINVAL => unreachable,
+        EDOM => return error.TimeoutTooBig,
+        EISCONN => return error.AlreadyConnected,
+        ENOPROTOOPT => return error.InvalidProtocolOption,
+        ENOTSOCK => return error.NotSocket,
```

Comment:
I'm pretty sure this would be a race condition, and therefore this should actually be `unreachable`. There's precedent for this in, for example, `std.os.bind`.